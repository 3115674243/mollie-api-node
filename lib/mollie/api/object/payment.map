{
  "version": 3,
  "file": "payment.js",
  "sourceRoot": "../../../..",
  "sources": [
    "src/lib/mollie/api/object/payment.coffee"
  ],
  "names": [],
  "mappings": ";AAAA;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;CAAA;CAAA;CAAA;CAAA,KAAA;;CAAA,CA8BA,CAAS,GAAT;CAAS,CAAK,CAAL,CAAA;CAAK,CAAQ,IAAR;MAAL;CA9BT,GAAA;;CAAA,CAgCA,CAAuB,GAAjB,CAAN;CACE,EAAoB,CAApB,EAAA,CAAC,IAAD;;CAAA,EACoB,CAApB,GAAC,IADD,KACA;;CADA,EAEoB,CAApB,GAAC,EAFD,KAEA;;CAFA,EAGoB,CAApB,EAHA,CAGC,IAAD;;CAEa,EAAA,CAAA,aAAA;CACX,CAAA,CAAM,CAAL,EAAD;CAAA,EACQ,CAAP,EAAD;CADA,EAEU,CAAT,EAAD;CAFA,EAGe,CAAd,EAAD,KAAA;CAHA,EAIU,CAAT,EAAD;CAJA,EAKU,CAAT,EAAD;CALA,EAMmB,CAAlB,EAAD,SAAA;CANA,EAOgB,CAAf,EAAD,MAAA;CAPA,EAQqB,CAApB,EAAD,WAAA;CARA,EASmB,CAAlB,EAAD,SAAA;CATA,EAUY,CAAX,EAAD,EAAA;CAVA,EAWW,CAAV,EAAD,CAAA;CAXA,EAaE,CADD,CAAD,CAAA;CACE,CAAY,EAAZ,IAAA,EAAA;CAAA,CACa,EADb,IACA,GAAA;CAfS,OACX;CANF,IAKa;;CALb,EAsBQ,GAAR,GAAQ;CACL,GAAA,CAAU,CAAX,KAAuB,EAAvB;CAvBF,IAsBQ;;CAtBR,EAyBQ,GAAR,GAAQ;AACL,CAAD,GAAG,SAAH;CA1BF,IAyBQ;;CAzBR,EA4Be,MAAA,IAAf;CACG,GAAA,CAAD,QAAA;CA7BF,IA4Be;;CA5Bf;;CAjCF;CAAA"
}