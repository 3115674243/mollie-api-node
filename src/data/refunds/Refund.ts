import { RefundData } from './data';
import refundHelpers from './helpers';
import Seal from '../../types/Seal';
import OrderLine, { injectPrototypes as injectOrderLinePrototypes } from '../orders/orderlines/OrderLine';

type Refund = Seal<RefundData & { lines?: OrderLine[] }, typeof refundHelpers>;

export default Refund;

export function injectPrototypes(input: RefundData): Refund {
  let lines: Refund['lines'];
  if (input.lines != undefined) {
    lines = input.lines.map(injectOrderLinePrototypes);
  }
  return Object.assign(Object.create(refundHelpers), input, {
    lines,
  });
}
