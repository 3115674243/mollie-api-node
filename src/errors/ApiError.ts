import { MollieApiError, MollieApiErrorLinks, Url } from '../data/global';
import { cloneDeep, get, has } from 'lodash';

/**
 * @since 3.0.0
 */
export default class ApiError extends Error {
  public constructor(message: string, protected title?: string, protected status?: number, protected field?: string, protected links?: MollieApiErrorLinks) {
    super(message);
    this.name = 'ApiError';
  }

  /**
   * Get the error message
   *
   * @returns The error message
   *
   * @since 3.0.0
   *
   * @public ✓ This method is part of the public API
   */
  public getMessage(): string {
    return this.message;
  }

  /**
   * Get the field name that contains an error
   *
   * @returns The error field
   *
   * @since 3.0.0
   *
   * @public ✓ This method is part of the public API
   */
  public getField(): string | undefined {
    return this.field;
  }

  /**
   * Get the API status code
   *
   * @returns The status code
   *
   * @since 3.0.0
   *
   * @public ✓ This method is part of the public API
   */
  public getStatusCode(): number | undefined {
    return this.status;
  }

  /**
   * Get the documentation URL
   *
   * @returns The documentation URL
   *
   * @since 3.0.0
   *
   * @public ✓ This method is part of the public API
   */
  public getDocumentationUrl(): string {
    return this.getUrl('documentation');
  }

  /**
   * Get the dashboard URL
   *
   * @returns The dashboard URL
   *
   * @since 3.0.0
   *
   * @public ✓ This method is part of the public API
   */
  public getDashboardUrl(): string {
    return this.getUrl('dashboard');
  }

  /**
   * Check if the link exists
   *
   * @param key - Link name
   *
   * @returns Whether the link exists
   *
   * @since 3.0.0
   *
   * @public ✓ This method is part of the public API
   */
  public hasLink(key: string): boolean {
    return has(this.links, key);
  }

  /***
   * Retrieve a link by name
   *
   * @param key - The link name
   *
   * @returns A link to the resource
   *
   * @since 3.0.0
   *
   * @public ✓ This method is part of the public API
   */
  public getLink(key: string): Url {
    return get(this.links, key);
  }

  /**
   *
   * @param {string} key
   *
   * @returns {string}
   *
   * @since 3.0.0
   *
   * @public ✓ This method is part of the public API
   */
  public getUrl(key: string): string {
    return get(this.getLink(key), 'href');
  }

  public toString(): string {
    return `ApiError: ${this.message}`;
  }

  /**
   * Create an `ApiError` from a raw error format
   *
   * @param error - A raw Mollie API error
   *
   * @returns A new `ApiError`
   *
   * @since 3.0.0
   */
  public static createFromResponse(error: MollieApiError): ApiError {
    return new ApiError(get(error, 'data.detail'), get(error, 'data.title'), get(error, 'data.status'), get(error, 'data.field'), cloneDeep(get(error, 'data._links')));
  }
}
