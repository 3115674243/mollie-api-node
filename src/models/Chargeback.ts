import Model from '../model';
import { IChargeback } from '../types/chargeback';

/**
 * The `Chargeback` model
 */
export default class Chargeback extends Model implements IChargeback {
  public static resourcePrefix = 'chb_';

  public resource = 'chargeback';
  public id = null;
  public amount = null;
  public settlementAmount = null;
  public createdAt = null;
  public reversedAt = null;
  public paymentId = null;
  public _links = {
    self: null,
    documentation: null,
    payment: null,
    settlement: null,
  };

  /**
   * Chargeback constructor
   *
   * @public âœ“ This method is part of the public API
   */
  constructor(props?: Partial<IChargeback>) {
    super(props);

    Object.assign(this, props);
  }
}
