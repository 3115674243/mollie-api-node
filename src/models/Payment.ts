import Model from '../model';
import { IPayment, PaymentStatus } from '../types/payment';
import { ApiMode, IAmount, ILinks, Locale, PaymentMethod } from '../types/global';

/**
 * The `Payment` model
 */
export default class Payment extends Model implements IPayment {
  public static readonly resourcePrefix = 'tr_';
  public resource: string;
  public id: string;

  public mode: ApiMode;
  public createdAt: string;
  public status: PaymentStatus;
  public isCancelable: boolean;
  public authorizedAt?: string;
  public paidAt?: string;
  public canceledAt?: string;
  public expiresAt: string;
  public expiredAt?: string;
  public failedAt?: string;
  public amount: IAmount;
  public amountRefunded?: IAmount;
  public amountRemaining?: IAmount;
  public amountCaptured?: IAmount;
  public description: string;
  public redirectUrl: string | null;
  public webhookUrl?: string;
  public method: PaymentMethod;
  public metadata: any;
  public locale: Locale;
  public countryCode?: string;
  public profileId: string;
  public settlementAmount?: IAmount;
  public settlementId?: string;
  public customerId?: string;
  public sequenceType: string;
  public mandateId?: string;
  public subscriptionId?: string;
  public orderId?: string;
  public details?: any;
  public applicationFee?: {
    amount: IAmount;
    description: string;
  };
  public _links: ILinks;

  constructor(props?: Partial<IPayment>) {
    super(props);

    const defaults: Partial<IPayment> = {
      resource: 'payment',
      id: null,
      mode: null,
      createdAt: null,
      status: null,
      isCancelable: null,
      paidAt: null,
      canceledAt: null,
      expiresAt: null,
      expiredAt: null,
      failedAt: null,
      amount: {
        value: null,
        currency: null,
      },
      amountRefunded: null,
      amountRemaining: null,
      description: null,
      redirectUrl: null,
      webhookUrl: null,
      method: null,
      metadata: null,
      locale: null,
      countryCode: null,
      profileId: null,
      settlementAmount: null,
      settlementId: null,
      customerId: null,
      sequenceType: null,
      mandateId: null,
      subscriptionId: null,
      applicationFee: {
        amount: {
          value: null,
          currency: null,
        },
        description: null,
      },
      details: null,
      _links: {
        checkout: null,
        refunds: null,
        chargebacks: null,
        settlement: null,
        mandate: null,
        subscription: null,
        customer: null,
      },
    };

    Object.assign(this, defaults, props);
  }

  /**
   * If the payment is open
   *
   * @returns {boolean}
   */
  isOpen() {
    return this.status === PaymentStatus.open;
  }

  /**
   * If the payment is authorized
   *
   * @returns {boolean}
   */
  isAuthorized() {
    return this.status === PaymentStatus.authorized;
  }

  /**
   * If the payment is paid
   *
   * @returns {boolean}
   */
  isPaid() {
    return !!this.paidAt;
  }

  /**
   * If the payment is canceled
   *
   * @returns {boolean}
   */
  isCanceled() {
    return !!this.canceledAt;
  }

  /**
   * If the payment is expired
   *
   * @returns {boolean}
   */
  isExpired() {
    return !!this.expiredAt;
  }

  /**
   * If the payment is refundable
   *
   * @returns {boolean}
   *
   * @since 2.0.0-rc.2
   */
  isRefundable() {
    return this.amountRemaining !== null;
  }

  /**
   * Get the payment URL
   *
   * @returns {string|null}
   */
  getPaymentUrl() {
    return this._links && this._links.checkout && this._links.checkout.href;
  }
}
