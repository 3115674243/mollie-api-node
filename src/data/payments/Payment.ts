import { PaymentData } from './data';
import Chargeback, { injectPrototypes as injectChargebackPrototypes } from '../chargebacks/Chargeback';
import Refund, { injectPrototypes as injectRefundPrototypes } from '../refunds/Refund';
import Seal from '../../types/Seal';
import paymentHelpers from './helpers';

type Payment = Seal<
  PaymentData & {
    _embedded?: {
      refunds?: Omit<Refund, '_embedded'>[];
      chargebacks?: Omit<Chargeback, '_embedded'>[];
    };
  },
  typeof paymentHelpers
>;

export default Payment;

export function injectPrototypes(input: PaymentData): Payment {
  let _embedded: Payment['_embedded'];
  if (input._embedded != undefined) {
    _embedded = {};
    if (input._embedded.chargebacks != undefined) {
      _embedded.chargebacks = input._embedded.chargebacks.map(injectChargebackPrototypes);
    }
    if (input._embedded.refunds != undefined) {
      _embedded.refunds = input._embedded.refunds.map(injectRefundPrototypes);
    }
  }
  return Object.assign(Object.create(paymentHelpers), input, { _embedded });
}
