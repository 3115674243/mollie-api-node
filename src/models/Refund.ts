import { isPlainObject } from 'lodash';

import Model from '../model';
import { RefundStatus } from '../types/refund';
import { IRefund } from '../types/payment/refund';
import Payment from './Payment';

/**
 * The `Refund` model
 *
 * {@link IRefund}
 */
export default class Refund extends Model implements IRefund {
  public static resourcePrefix = 're_';

  public resource = 'refund';
  public id = null;
  public amount = {
    currency: null,
    value: null,
  };
  public settlementAmount = null;
  public description = null;
  public status = null;
  public createdAt = null;
  public paymentId = null;
  public _links = {
    payment: null,
    settlement: null,
    order: null,
    self: null,
    documentation: null,
  };
  public lines = null;
  public _embedded = null;
  public orderId = null;

  /**
   * Refund constructor
   *
   * @public ✓ This constructor is part of the public API
   */
  public constructor(props?: Partial<IRefund>) {
    super();

    Object.assign(this, props);

    if (this._embedded != null && isPlainObject(this._embedded)) {
      if (Array.isArray(this._embedded.payments)) {
        // eslint-disable-next-line array-callback-return
        this._embedded.payments.map((payment, key, payments) => {
          // eslint-disable-next-line no-param-reassign
          payments[key] = new Payment(payment);
        });
      }
    }
  }

  /**
   * The refund is queued until there is enough balance to process te refund.
   * You can still cancel the refund.
   *
   * @returns {boolean}
   *
   * @public ✓ This method is part of the public API
   */
  public isQueued(): boolean {
    return this.status === RefundStatus.queued;
  }

  /**
   * The refund will be sent to the bank on the next business day. You can still cancel the refund.
   *
   * @public ✓ This method is part of the public API
   */
  public isPending(): boolean {
    return this.status === RefundStatus.pending;
  }

  /**
   * The refund has been sent to the bank. The refund amount will be transferred to the consumer
   * account as soon as possible.
   *
   * @public ✓ This method is part of the public API
   */
  public isProcessing(): boolean {
    return this.status === RefundStatus.processing;
  }

  /**
   * The refund amount has been transferred to the consumer.
   *
   * @public ✓ This method is part of the public API
   */
  public isRefunded(): boolean {
    return this.status === RefundStatus.refunded;
  }

  /**
   * The refund has failed during processing.
   *
   * @public ✓ This method is part of the public API
   */
  public isFailed(): boolean {
    return this.status === RefundStatus.failed;
  }
}
