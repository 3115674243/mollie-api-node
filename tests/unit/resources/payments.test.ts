import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import PaymentsResource from '../../../src/resources/payments';
import Payment from '../../../src/models/Payment';
import paymentsResponse from '../__stubs__/payments.json';
import paymentsEmbedChargebackResponse from '../__stubs__/payments_embed_chargebacks.json';
import paymentsEmbedsResponse from '../__stubs__/payments_embeds.json';
import ApiError from '../../../src/errors/ApiError';
import Chargeback from '../../../src/models/Chargeback';
import Refund from '../../../src/models/Refund';
import { PaymentMethod } from '../../../src/types/global';
import { PaymentEmbed } from '../../../src/types/payment';

const mock = new MockAdapter(axios);

const props = {
  id: 'tr_nBeryjMVjr',
  amount: {
    currency: 'GBP',
    value: '75.00',
  },
  description: 'Test payment',
  method: PaymentMethod.ideal,
  metadata: {
    orderId: '12345',
  },
};

describe('payments', () => {
  let payments: PaymentsResource;
  beforeEach(() => {
    payments = new PaymentsResource(axios.create());
  });

  it('should have a resource name and model', () => {
    expect(PaymentsResource.resource).toBe('payments');
    expect(PaymentsResource.model).toBe(Payment);
  });

  describe('.create()', () => {
    const error = {
      field: 'amount',
      detail: 'The amount is lower than the minimum',
    };

    mock
      .onPost('/payments', {
        ...props,
        amount: { value: '0.05', currency: 'EUR' },
      })
      .reply(500, error);
    mock.onPost('/payments').reply(200, paymentsResponse._embedded.payments[0]);
    mock.onPost('/payments?include=details.qrCode').reply(200, paymentsResponse._embedded.payments[1]);

    it('should return a payment instance', () =>
      payments.create(props).then(result => {
        expect(result).toBeInstanceOf(Payment);
        expect(result.amount.value).toBe(props.amount.value);
        expect(result).toMatchSnapshot();
      }));

    it('should work with a callback', done => {
      payments.create(props, (err, result) => {
        expect(err).toBeNull();
        expect(result).toBeInstanceOf(Payment);
        expect(result.amount.value).toBe(props.amount.value);
        expect(result).toMatchSnapshot();
        done();
      });
    });

    it('should fail with a unsupported amount', done =>
      payments
        .create({
          ...props,
          amount: { value: '0.05', currency: 'EUR' },
        })
        .then(result => expect(result).toBeUndefined())
        .catch(err => {
          expect(err).toBeInstanceOf(ApiError);
          expect(err.getMessage()).toEqual(error.detail);
          expect(err.getField()).toBe(error.field);
          done();
        }));

    it('should return a QR code', done =>
      payments
        .create({
          ...props,
          include: 'details.qrCode',
        })
        .then(result => {
          expect(result).toBeInstanceOf(Payment);
          expect(result.amount.value).toBe(props.amount.value);
          expect(result.details).toBeDefined();
          expect((result.details as any).qrCode.width).toBe(180);
          expect(result).toMatchSnapshot();
          done();
        }));

    it('should return a QR code (alternative with array)', done =>
      payments
        .create({
          ...props,
          include: ['details.qrCode'],
        })
        .then(result => {
          expect(result).toBeInstanceOf(Payment);
          expect(result.amount.value).toBe(props.amount.value);
          expect(result.details).toBeDefined();
          expect((result.details as any).qrCode.width).toBe(180);
          expect(result).toMatchSnapshot();
          done();
        }));
  });

  describe('.get()', () => {
    const error = { detail: 'The payment id is invalid' };

    mock.onGet(`/payments/${props.id}`).reply(200, paymentsResponse._embedded.payments[0]);
    mock.onGet(`/payments/${props.id}?embed=chargebacks`).reply(200, paymentsEmbedChargebackResponse._embedded.payments[0]);
    mock.onGet(`/payments/${props.id}?embed=chargebacks%3Brefunds`).reply(200, paymentsEmbedsResponse._embedded.payments[0]);
    mock.onGet('/payments/foo').reply(500, error);

    it('should return a payment instance', () =>
      payments.get(props.id).then(result => {
        expect(result).toBeInstanceOf(Payment);
        expect(result).toMatchSnapshot();
      }));

    it('should work with a callback', done => {
      payments.get(props.id, (err, result) => {
        expect(err).toBeNull();
        expect(result).toBeInstanceOf(Payment);
        expect(result).toMatchSnapshot();
        done();
      });
    });

    it('should return an error for non-existing IDs', done =>
      payments
        .get('foo')
        .then(result => expect(result).toBeUndefined())
        .catch(err => {
          expect(err).toBeInstanceOf(ApiError);
          done();
        }));

    it('should return an error with a callback for non-existing IDs', done => {
      payments.get('foo', (err, result) => {
        expect(err).toBeInstanceOf(ApiError);
        expect(err.getMessage()).toEqual(error.detail);
        expect(result).toBeUndefined();
        done();
      });
    });

    it('should return a chargeback embed', done => {
      payments
        .get(props.id, { embed: PaymentEmbed.chargebacks })
        .then(result => {
          expect(result).toBeInstanceOf(Payment);
          expect(result._embedded.chargebacks[0]).toBeInstanceOf(Chargeback);
          done();
        })
        .catch(err => expect(err).toBeUndefined());
    });

    it('should return a chargeback embed (alternative with an array)', done => {
      payments
        .get(props.id, { embed: [PaymentEmbed.chargebacks] })
        .then(result => {
          expect(result).toBeInstanceOf(Payment);
          expect(result._embedded.chargebacks[0]).toBeInstanceOf(Chargeback);
          done();
        })
        .catch(err => expect(err).toBeUndefined());
    });

    it('should return multiple embeds', done => {
      payments
        .get(props.id, { embed: [PaymentEmbed.chargebacks, PaymentEmbed.refunds] })
        .then(result => {
          expect(result).toBeInstanceOf(Payment);
          expect(result._embedded.chargebacks[0]).toBeInstanceOf(Chargeback);
          expect(result._embedded.refunds[0]).toBeInstanceOf(Refund);
          done();
        })
        .catch(err => expect(err).toBeUndefined());
    });
  });

  describe('.all()', () => {
    mock.onGet('/payments').reply(200, paymentsResponse);

    it('should return a list of all payments', () =>
      payments.all().then(result => {
        expect(result).toBeInstanceOf(Array);
        expect(result).toHaveProperty('links');
        expect(result).toMatchSnapshot();
      }));

    it('should work with a callback', done => {
      payments.all((err, result) => {
        expect(err).toBeNull();
        expect(result).toBeInstanceOf(Array);
        expect(result).toHaveProperty('links');
        expect(result).toMatchSnapshot();
        done();
      });
    });
  });

  describe('.cancel()', () => {
    const error = { detail: 'Method not allowed' };

    mock.onDelete(`/payments/expired`).reply(500, error);
    mock.onDelete(`/payments/${props.id}`).reply(200, paymentsResponse._embedded.payments[0]);

    it('should return the canceled payment when it could be canceled', done =>
      payments
        .cancel(props.id)
        .then(result => {
          expect(result).toMatchObject(paymentsResponse._embedded.payments[0]);
          expect(result).toMatchSnapshot();
          done();
        })
        .catch(err => expect(err).toBeUndefined()));

    it('should return the status when the payment could not be canceled', done => {
      payments
        .cancel('expired')
        .then(result => expect(result).toBeUndefined())
        .catch(err => {
          expect(err).toBeInstanceOf(ApiError);
          done();
        });
    });
  });
});
