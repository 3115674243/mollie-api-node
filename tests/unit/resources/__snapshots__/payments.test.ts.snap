// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`payments .all() should return a list of all payments 1`] = `
Array [
  Payment {
    "_links": Object {
      "checkout": Object {
        "href": "https://www.mollie.com/paymentscreen/issuer/select/ideal/7UhSN1zuXS",
        "type": "text/html",
      },
      "self": Object {
        "href": "https://api.mollie.com/v2/payments/tr_7UhSN1zuXS",
        "type": "application/hal+json",
      },
    },
    "amount": Object {
      "currency": "GBP",
      "value": "75.00",
    },
    "createdAt": "2018-02-12T11:58:35.0Z",
    "description": "test",
    "details": null,
    "expiresAt": "2018-02-12T12:13:35.0Z",
    "id": "tr_7UhSN1zuXS",
    "isCancelable": false,
    "metadata": null,
    "method": "ideal",
    "mode": "test",
    "profileId": "pfl_QkEhN94Ba",
    "redirectUrl": "https://webshop.example.org/order/12345/",
    "resource": "payment",
    "status": "open",
  },
  Payment {
    "_links": Object {
      "checkout": Object {
        "href": "https://www.mollie.com/paymentscreen/issuer/select/ideal/7UhSN1zuXS",
        "type": "text/html",
      },
      "self": Object {
        "href": "https://api.mollie.com/v2/payments/tr_7UhSN1zuXS",
        "type": "application/hal+json",
      },
    },
    "amount": Object {
      "currency": "GBP",
      "value": "75.00",
    },
    "createdAt": "2018-02-12T11:58:35.0Z",
    "data": Object {
      "_links": Object {
        "checkout": Object {
          "href": "https://www.mollie.com/paymentscreen/issuer/select/ideal/7UhSN1zuXS",
          "type": "text/html",
        },
        "self": Object {
          "href": "https://api.mollie.com/v2/payments/tr_7UhSN1zuXS",
          "type": "application/hal+json",
        },
      },
      "amount": Object {
        "currency": "GBP",
        "value": "75.00",
      },
      "createdAt": "2018-02-12T11:58:35.0Z",
      "description": "test",
      "details": Object {
        "qrCode": Object {
          "height": 180,
          "src": "https://qr2.ideal.nl/ideal-qr/qr/get/3d530c37-e2c0-4f8b-b40c-d72a4bef40c2",
          "width": 180,
        },
      },
      "expiresAt": "2018-02-12T12:13:35.0Z",
      "id": "tr_7UhSN1zuXS",
      "isCancelable": false,
      "metadata": null,
      "method": "ideal",
      "mode": "test",
      "profileId": "pfl_QkEhN94Ba",
      "redirectUrl": "https://webshop.example.org/order/12345/",
      "resource": "payment",
      "status": "open",
    },
    "description": "test",
    "details": Object {
      "qrCode": Object {
        "height": 180,
        "src": "https://qr2.ideal.nl/ideal-qr/qr/get/3d530c37-e2c0-4f8b-b40c-d72a4bef40c2",
        "width": 180,
      },
    },
    "expiresAt": "2018-02-12T12:13:35.0Z",
    "id": "tr_7UhSN1zuXS",
    "isCancelable": false,
    "metadata": null,
    "method": "ideal",
    "mode": "test",
    "profileId": "pfl_QkEhN94Ba",
    "redirectUrl": "https://webshop.example.org/order/12345/",
    "resource": "payment",
    "status": "open",
  },
]
`;

exports[`payments .all() should work with a callback 1`] = `
Array [
  Payment {
    "_links": Object {
      "checkout": Object {
        "href": "https://www.mollie.com/paymentscreen/issuer/select/ideal/7UhSN1zuXS",
        "type": "text/html",
      },
      "self": Object {
        "href": "https://api.mollie.com/v2/payments/tr_7UhSN1zuXS",
        "type": "application/hal+json",
      },
    },
    "amount": Object {
      "currency": "GBP",
      "value": "75.00",
    },
    "createdAt": "2018-02-12T11:58:35.0Z",
    "description": "test",
    "details": null,
    "expiresAt": "2018-02-12T12:13:35.0Z",
    "id": "tr_7UhSN1zuXS",
    "isCancelable": false,
    "metadata": null,
    "method": "ideal",
    "mode": "test",
    "profileId": "pfl_QkEhN94Ba",
    "redirectUrl": "https://webshop.example.org/order/12345/",
    "resource": "payment",
    "status": "open",
  },
  Payment {
    "_links": Object {
      "checkout": Object {
        "href": "https://www.mollie.com/paymentscreen/issuer/select/ideal/7UhSN1zuXS",
        "type": "text/html",
      },
      "self": Object {
        "href": "https://api.mollie.com/v2/payments/tr_7UhSN1zuXS",
        "type": "application/hal+json",
      },
    },
    "amount": Object {
      "currency": "GBP",
      "value": "75.00",
    },
    "createdAt": "2018-02-12T11:58:35.0Z",
    "data": Object {
      "_links": Object {
        "checkout": Object {
          "href": "https://www.mollie.com/paymentscreen/issuer/select/ideal/7UhSN1zuXS",
          "type": "text/html",
        },
        "self": Object {
          "href": "https://api.mollie.com/v2/payments/tr_7UhSN1zuXS",
          "type": "application/hal+json",
        },
      },
      "amount": Object {
        "currency": "GBP",
        "value": "75.00",
      },
      "createdAt": "2018-02-12T11:58:35.0Z",
      "description": "test",
      "details": Object {
        "qrCode": Object {
          "height": 180,
          "src": "https://qr2.ideal.nl/ideal-qr/qr/get/3d530c37-e2c0-4f8b-b40c-d72a4bef40c2",
          "width": 180,
        },
      },
      "expiresAt": "2018-02-12T12:13:35.0Z",
      "id": "tr_7UhSN1zuXS",
      "isCancelable": false,
      "metadata": null,
      "method": "ideal",
      "mode": "test",
      "profileId": "pfl_QkEhN94Ba",
      "redirectUrl": "https://webshop.example.org/order/12345/",
      "resource": "payment",
      "status": "open",
    },
    "description": "test",
    "details": Object {
      "qrCode": Object {
        "height": 180,
        "src": "https://qr2.ideal.nl/ideal-qr/qr/get/3d530c37-e2c0-4f8b-b40c-d72a4bef40c2",
        "width": 180,
      },
    },
    "expiresAt": "2018-02-12T12:13:35.0Z",
    "id": "tr_7UhSN1zuXS",
    "isCancelable": false,
    "metadata": null,
    "method": "ideal",
    "mode": "test",
    "profileId": "pfl_QkEhN94Ba",
    "redirectUrl": "https://webshop.example.org/order/12345/",
    "resource": "payment",
    "status": "open",
  },
]
`;

exports[`payments .create() should return a QR code 1`] = `
Payment {
  "_links": Object {
    "checkout": Object {
      "href": "https://www.mollie.com/paymentscreen/issuer/select/ideal/7UhSN1zuXS",
      "type": "text/html",
    },
    "self": Object {
      "href": "https://api.mollie.com/v2/payments/tr_7UhSN1zuXS",
      "type": "application/hal+json",
    },
  },
  "amount": Object {
    "currency": "GBP",
    "value": "75.00",
  },
  "createdAt": "2018-02-12T11:58:35.0Z",
  "data": Object {
    "_links": Object {
      "checkout": Object {
        "href": "https://www.mollie.com/paymentscreen/issuer/select/ideal/7UhSN1zuXS",
        "type": "text/html",
      },
      "self": Object {
        "href": "https://api.mollie.com/v2/payments/tr_7UhSN1zuXS",
        "type": "application/hal+json",
      },
    },
    "amount": Object {
      "currency": "GBP",
      "value": "75.00",
    },
    "createdAt": "2018-02-12T11:58:35.0Z",
    "description": "test",
    "details": Object {
      "qrCode": Object {
        "height": 180,
        "src": "https://qr2.ideal.nl/ideal-qr/qr/get/3d530c37-e2c0-4f8b-b40c-d72a4bef40c2",
        "width": 180,
      },
    },
    "expiresAt": "2018-02-12T12:13:35.0Z",
    "id": "tr_7UhSN1zuXS",
    "isCancelable": false,
    "metadata": null,
    "method": "ideal",
    "mode": "test",
    "profileId": "pfl_QkEhN94Ba",
    "redirectUrl": "https://webshop.example.org/order/12345/",
    "resource": "payment",
    "status": "open",
  },
  "description": "test",
  "details": Object {
    "qrCode": Object {
      "height": 180,
      "src": "https://qr2.ideal.nl/ideal-qr/qr/get/3d530c37-e2c0-4f8b-b40c-d72a4bef40c2",
      "width": 180,
    },
  },
  "expiresAt": "2018-02-12T12:13:35.0Z",
  "id": "tr_7UhSN1zuXS",
  "isCancelable": false,
  "metadata": null,
  "method": "ideal",
  "mode": "test",
  "profileId": "pfl_QkEhN94Ba",
  "redirectUrl": "https://webshop.example.org/order/12345/",
  "resource": "payment",
  "status": "open",
}
`;

exports[`payments .create() should return a payment instance 1`] = `
Payment {
  "_links": Object {
    "checkout": Object {
      "href": "https://www.mollie.com/paymentscreen/issuer/select/ideal/7UhSN1zuXS",
      "type": "text/html",
    },
    "self": Object {
      "href": "https://api.mollie.com/v2/payments/tr_7UhSN1zuXS",
      "type": "application/hal+json",
    },
  },
  "amount": Object {
    "currency": "GBP",
    "value": "75.00",
  },
  "createdAt": "2018-02-12T11:58:35.0Z",
  "description": "test",
  "details": null,
  "expiresAt": "2018-02-12T12:13:35.0Z",
  "id": "tr_7UhSN1zuXS",
  "isCancelable": false,
  "metadata": null,
  "method": "ideal",
  "mode": "test",
  "profileId": "pfl_QkEhN94Ba",
  "redirectUrl": "https://webshop.example.org/order/12345/",
  "resource": "payment",
  "status": "open",
}
`;

exports[`payments .create() should work with a callback 1`] = `
Payment {
  "_links": Object {
    "checkout": Object {
      "href": "https://www.mollie.com/paymentscreen/issuer/select/ideal/7UhSN1zuXS",
      "type": "text/html",
    },
    "self": Object {
      "href": "https://api.mollie.com/v2/payments/tr_7UhSN1zuXS",
      "type": "application/hal+json",
    },
  },
  "amount": Object {
    "currency": "GBP",
    "value": "75.00",
  },
  "createdAt": "2018-02-12T11:58:35.0Z",
  "description": "test",
  "details": null,
  "expiresAt": "2018-02-12T12:13:35.0Z",
  "id": "tr_7UhSN1zuXS",
  "isCancelable": false,
  "metadata": null,
  "method": "ideal",
  "mode": "test",
  "profileId": "pfl_QkEhN94Ba",
  "redirectUrl": "https://webshop.example.org/order/12345/",
  "resource": "payment",
  "status": "open",
}
`;

exports[`payments .get() should return a payment instance 1`] = `
Payment {
  "_links": Object {
    "checkout": Object {
      "href": "https://www.mollie.com/paymentscreen/issuer/select/ideal/7UhSN1zuXS",
      "type": "text/html",
    },
    "self": Object {
      "href": "https://api.mollie.com/v2/payments/tr_7UhSN1zuXS",
      "type": "application/hal+json",
    },
  },
  "amount": Object {
    "currency": "GBP",
    "value": "75.00",
  },
  "createdAt": "2018-02-12T11:58:35.0Z",
  "description": "test",
  "details": null,
  "expiresAt": "2018-02-12T12:13:35.0Z",
  "id": "tr_7UhSN1zuXS",
  "isCancelable": false,
  "metadata": null,
  "method": "ideal",
  "mode": "test",
  "profileId": "pfl_QkEhN94Ba",
  "redirectUrl": "https://webshop.example.org/order/12345/",
  "resource": "payment",
  "status": "open",
}
`;

exports[`payments .get() should work with a callback 1`] = `
Payment {
  "_links": Object {
    "checkout": Object {
      "href": "https://www.mollie.com/paymentscreen/issuer/select/ideal/7UhSN1zuXS",
      "type": "text/html",
    },
    "self": Object {
      "href": "https://api.mollie.com/v2/payments/tr_7UhSN1zuXS",
      "type": "application/hal+json",
    },
  },
  "amount": Object {
    "currency": "GBP",
    "value": "75.00",
  },
  "createdAt": "2018-02-12T11:58:35.0Z",
  "description": "test",
  "details": null,
  "expiresAt": "2018-02-12T12:13:35.0Z",
  "id": "tr_7UhSN1zuXS",
  "isCancelable": false,
  "metadata": null,
  "method": "ideal",
  "mode": "test",
  "profileId": "pfl_QkEhN94Ba",
  "redirectUrl": "https://webshop.example.org/order/12345/",
  "resource": "payment",
  "status": "open",
}
`;
